src/renderers/dom/shared/__tests__/ReactDOMAttribute-test.js

```diff
@@ -0,0 +1,125 @@
+    });
+
+    it('removes unknown attributes that were rendered but are now missing', () => {
+      var el = document.createElement('div');
+      spyOn(console, 'error');
+      ReactDOM.render(<div unknown="something" />, el);
+      expect(el.firstChild.getAttribute('unknown')).toBe('something');
+      expectDev(console.error.calls.count(0)).toBe(0);
+      ReactDOM.render(<div />, el);
+      expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+      expectDev(console.error.calls.count(0)).toBe(0);
+    });
+
+    it('passes through strings to unknown attributes', () => {
+      var el = document.createElement('div');
+      spyOn(console, 'error');
+      ReactDOM.render(<div unknown="something" />, el);
+      expect(el.firstChild.getAttribute('unknown')).toBe('something');
+      expectDev(console.error.calls.count(0)).toBe(0);
+      ReactDOM.render(<div />, el);
+      expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+      expectDev(console.error.calls.count(0)).toBe(0);
+    });
+
+    it('coerces unknown attributes to strings with numbers and booleans', () => {
+      var el = document.createElement('div');
+      spyOn(console, 'error');
+
+      function testCoerceToString(value) {
+        ReactDOM.render(<div unknown="something" />, el);
+        expect(el.firstChild.getAttribute('unknown')).toBe('something');
+        expectDev(console.error.calls.count(0)).toBe(0);
+        ReactDOM.render(<div unknown={value} />, el);
+        expect(el.firstChild.getAttribute('unknown')).toBe(value + '');
+        expectDev(console.error.calls.count(0)).toBe(0);
+        console.error.calls.reset();
+      }
+
+      testCoerceToString(0);
+      testCoerceToString(-1);
+      testCoerceToString(42);
+      testCoerceToString(9000.99999);
+      // TODO: either change what we expect here or update the implementation
+      // so that these pass -
+      //
+      // testCoerceToString(true);
+      // testCoerceToString(false);
+    });
+
+    // TODO: get this test passing
+    xit(
+      'coerces unknown attributes to strings **and warns** with NaN, symbols, functions, and objects',
+      () => {
+        var el = document.createElement('div');
+        spyOn(console, 'error');
+
+        function testCoerceToString(value) {
+          ReactDOM.render(<div unknown="something" />, el);
+          expect(el.firstChild.getAttribute('unknown')).toBe('something');
+          expectDev(console.error.calls.count(0)).toBe(0);
+          ReactDOM.render(<div unknown={value} />, el);
+          expect(el.firstChild.getAttribute('unknown')).toBe(value + '');
+          expectDev(console.error.calls.count(0)).toBe(1);
+          // TODO: add specific expectations about what the warning says
+          // expectDev(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(...
+          console.error.calls.reset();
+        }
+
+        // TODO: this does not warn. We think it should.
+        testCoerceToString(NaN);
```

Comment:
This is curious because [we check for NaN](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js#L106-L116). Is it possible that the UnknownPropertyHook is caching a warning for the `unknown` prop?